src/BloomExe/WebLibraryIntegration/BloomS3Client.cs

```diff
@@ -355,13 +355,29 @@ public string DownloadFile(string storageKeyOfFile)
 		/// <param name="storageKeyOfBookFolder"></param>
 		public string DownloadBook(string storageKeyOfBookFolder, string pathToDestinationParentDirectory, ProgressDialog downloadProgress = null)
 		{
-			//TODO tell it not to download pdfs. Those are just in there for previewing purposes, we don't need to get them now that we're getting the real thing
-
 			//review: should we instead save to a newly created folder so that we don't have to worry about the
 			//other folder existing already? Todo: add a test for that first.
 
-			if (!GetBookExists(storageKeyOfBookFolder))
+			// We need to download individual files to avoid downloading unwanted files (PDFs and thumbs.db to
+			// be specific).  See https://silbloom.myjetbrains.com/youtrack/issue/BL-2312.  So we need the list
+			// of items, not just the count.
+			var matching = GetMatchingItems(storageKeyOfBookFolder);
+			int totalItems = 0;
+			string folderName = null;
+			for (int i = 0; i < matching.S3Objects.Count; ++i)
+			{
+				var objKey = matching.S3Objects[i].Key;
+				if (String.IsNullOrEmpty(folderName))
+					folderName = Path.GetFileName(Path.GetDirectoryName(objKey));
+				var filename = Path.GetFileName(objKey);
+				if (filename.ToLowerInvariant() == "thumbs.db" || filename.ToLowerInvariant().EndsWith(".pdf"))
```

Comment:
It would be good to have a single list of things to filter out; else we may make a change in one of these places and not the other.